src/sentry/models/sentryapp.py

```diff
@@ -153,3 +155,7 @@ def build_signature(self, body):
         return hmac.new(
             key=secret.encode("utf-8"), msg=body.encode("utf-8"), digestmod=sha256
         ).hexdigest()
+
+    def show_auth_info(self, access):
+        encoded_scopes = set({u"%s" % scope for scope in list(access.scopes)})
+        return set(self.scope_list).issubset(encoded_scopes)
```

Comment:
Not sure if this is relevant but I noticed that in https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/sentryapps.py#L48 the permissions are calculated slightly differently in that having any of the required scopes allows the request -- should this be the same or is this just different?

Reply:
This is a bit different because we want to make sure the user has all the same (if not more) permissions as the application. If the app had the scopes `['org:read', 'org:write']` and the user had just `['org:read']` we still don't want them to be able to use a token with the `org:write` permissions.